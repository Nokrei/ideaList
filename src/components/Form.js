import React, { useState } from "react";
import PropTypes from "prop-types";
import { useForm } from "react-hook-form";
import styles from "../styles/Form.module.css";

export default function Form({ submitForm }) {
  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm();

  const maxLength = 140;
  const count = watch("description");

  const onSubmit = submitForm;

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <div className={styles.inputGroup}>
        <label htmlFor="title">Title</label>
        <input
          id="title"
          defaultValue="test"
          {...register("title", { required: true, maxLength: 20 })}
        />
        {errors.title && <span>This field is required</span>}
      </div>
      <div className={styles.inputGroup}>
        <label htmlFor="description">Description</label>
        <p>
          {typeof count === "string" ? count.length : 0} / {maxLength}
        </p>
        <textarea
          id="description"
          rows={10}
          {...register("description", { required: true, maxLength: maxLength })}
        />
        {errors.description && <span>This field is required</span>}
      </div>

      <input type="submit" />
    </form>
  );
}

Form.propTypes = {
  submitForm: PropTypes.func.isRequired,
};
