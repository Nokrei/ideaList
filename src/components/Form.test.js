import React from "react";
import { render } from "@testing-library/react";
import Form from "./Form";

describe("Form component", () => {
  it("should render", () => {
    const { container } = render(
      <Form submitForm={() => console.log("submitted")} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
