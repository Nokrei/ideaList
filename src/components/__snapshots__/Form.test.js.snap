// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render 1`] = `
<form>
  <div
    class="inputGroup"
  >
    <label
      for="title"
    >
      Title
    </label>
    <input
      id="title"
      name="title"
    />
  </div>
  <div
    class="inputGroup"
  >
    <label
      for="description"
    >
      Description
    </label>
    <p>
      0
       / 
      140
    </p>
    <textarea
      id="description"
      name="description"
      rows="10"
    />
  </div>
  <input
    class="btn btn-primary"
    type="submit"
  />
</form>
`;
